ext/intl/tests/bug76093.phpt

```diff
@@ -0,0 +1,75 @@
+--TEST--
+Bug #76093 (NumberFormatter::format loses precision)
+--SKIPIF--
+<?php if( !extension_loaded( 'intl' ) ) print 'skip'; ?>
+--FILE--
+<?php
+
+# See also https://phabricator.wikimedia.org/T268456
+$x = new NumberFormatter('en_US', NumberFormatter::DECIMAL);
+$x2 = new NumberFormatter('en_US', NumberFormatter::CURRENCY);
+foreach ([
+    '999999999999999999', # Fits in signed 64-bit integer
+    '9999999999999999999', # Does not fit in signed 64-bit integer
+    9999999999999999999, # Precision loss seen when passing as number
+    ] as $value) {
+    try {
+        var_dump([
+            'input' => $value,
+            'default' => $x->format($value),
+            'int64' => $x->format($value, NumberFormatter::TYPE_INT64),
+            'double' => $x->format($value, NumberFormatter::TYPE_DOUBLE),
+            'decimal' => $x->format($value, NumberFormatter::TYPE_DECIMAL),
+            # formatCurrency requires the NumberFormatter to be created with
+            # the CURRENCY or CURRENCY_ACCOUNTING style.
+            'currency' => $x2->formatCurrency($value, 'USD'),
+        ]);
+    } catch (TypeError $ex) {
+        echo $ex->getMessage(), PHP_EOL;
+    }
+}
+
+?>
+--EXPECT--
+array(6) {
+  ["input"]=>
+  string(18) "999999999999999999"
+  ["default"]=>
+  string(23) "999,999,999,999,999,999"
+  ["int64"]=>
+  string(23) "999,999,999,999,999,999"
```

Comment:
[This breaks on 32bit architectures](https://dev.azure.com/phpazuredevops/PHP/_build/results?buildId=22263&view=logs&j=d08b4992-9a38-5fb7-07af-0ddc5109f80d&t=6680665c-818b-512c-90ed-d46d91df45b0&l=917). Not sure how to solve this best. Maybe have separate test cases for 64/32bit, or maybe use a `%r...%r` format sequence to match either result.

Reply:
It's somewhat surprising to me that the `NumberFormatter::TYPE_INT64` type isn't actually guaranteed to be a 64-bit int, but I guess that's because php's zval can't actually hold a 64-bit int.  Is the `%r` syntax documented anywhere?

Reply:
Ok, latest push should resolve this (using %r for the numeric case, and TYPE_DECIMAL for the string case).